### PR TITLE
Code clean up for CoreUtils and VarDeclNode

### PR DESCRIPTION
**Pull request by ke-yu**
### Purpose

Remove unnecessary parameter for some functions in `ProtoCore.CoreUtils`: change 

``` 
public static void InsertPredefinedAndBuiltinMethods(Core core, ProtoCore.AST.Node root, bool builtinMethodsLoaded)
{
    InsertPredefinedMethod(core, root, builtinMethodsLoaded);
   ...
}
public static void InsertPredefinedMethod(Core core, ProtoCore.AST.Node root, bool builtinMethodsLoaded)
{
    if (!builtinMethodsLoaded) {
        ...
        (root as CodeBlockNode)...
    }
}
```
to
```
public static void InsertPredefinedAndBuiltinMethod(Core core, CodeBlockNode root)
{
    InsertPredefinedMethod(core, root);
}
public static void InsertPredefinedMethod(Core core, CodeBlockNode root)
{
    ...
}
```

Remove `VarDeclNode.Attrbiutes` and `VarDeclNode.memoryregion` properties; rename `VarDeclNode.access` to `VarDeclNode.Access`.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.

### Reviewers

@sharadkjaiswal

### CODE REVIEW

File: src/Engine/ProtoAssociative/CodeGen.cs [hunk 1]
```diff
@@ -5164,8 +5163,6 @@ private void EmitMemberVariables(ClassDeclNode classDecl, GraphNode graphNode =
                                 currentClassScope = globalClassIndex;
                                 ix = 0;
                             }
-                            ProtoCore.AST.AssociativeAST.VarDeclNode varnode = classDecl.varlist[ix++] as ProtoCore.AST.AssociativeAST.VarDeclNode;
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoAssociative/CodeGen.cs]: Why are these lines removed? How is `ix` being incremented now?

File: src/Engine/ProtoCore/Parser/AssociativeAST.cs [hunk 2]
```diff
@@ -1033,16 +1029,14 @@ public VarDeclNode(VarDeclNode rhs)
                 Name = rhs.ArgumentType.Name
             };
             NameNode = NodeUtils.Clone(rhs.NameNode);
-            access = rhs.access;
+            Access = rhs.Access;
             IsStatic = rhs.IsStatic;
             ExternalAttributes = rhs.ExternalAttributes;
         }
 
-        public List<AssociativeNode> Attributes { get; set; }
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Parser/AssociativeAST.cs]: Are we removing support of attribute on variables? Is it because we are not supporting class definition in DS now?

**Comment by sharadkjaiswal** [comment]
Apart from above comments, its LGTM.

File: src/Engine/ProtoCore/Parser/AssociativeAST.cs (hunk 2 above)
Comment (ke-yu) [inline src/Engine/ProtoCore/Parser/AssociativeAST.cs]: This field never used in the code.
